client/src/templates/Challenges/classic/lower-jaw.tsx

```diff
@@ -178,97 +238,90 @@ const LowerJaw = ({
-              className='btn fade-in'
-              id='get-help-button'
-              title={t('buttons.get-help')}
-              aria-label={t('buttons.get-help')}
-              data-cy='get-help-button'
-              onClick={openHelpModal}
-            >
-              <Help />
-            </button>
-          ) : null}
-        </div>
-      </div>
-    );
-  };
+  const isAttemptsLargerThanTest =
+    currentAttempts &&
+    testsLength &&
+    (currentAttempts >= testsLength || currentAttempts >= 3);
 
   const showDesktopButton = window.innerWidth > MAX_MOBILE_WIDTH;
 
-  const renderButtons = () => {
-    return (
-      <>
-        <div id='action-buttons-container'>
-          {isSignedIn ? null : challengeIsCompleted ? (
-            <Button
-              block={true}
-              href={`${apiLocation}/signin`}
-              className='btn-cta'
-            >
-              {t('learn.sign-in-save')}
-            </Button>
-          ) : null}
-          <button
-            id='test-button'
-            data-cy='run-tests-button'
-            className={`btn-block btn ${challengeIsCompleted ? 'sr-only' : ''}`}
-            aria-hidden={testBtnAriaHidden}
-            onClick={tryToExecuteChallenge}
-          >
-            {showDesktopButton
-              ? t('buttons.check-code')
-              : t('buttons.check-code-2')}
-          </button>
-          <button
-            id='submit-button'
-            data-cy='submit-button'
-            aria-hidden={!challengeIsCompleted}
-            className='btn-block btn'
-            onClick={tryToSubmitChallenge}
-            ref={submitButtonRef}
-          >
-            {t('buttons.submit-and-go')}
-          </button>
-        </div>
-      </>
-    );
-  };
+  const checkButtonText = showDesktopButton
+    ? t('buttons.check-code')
+    : t('buttons.check-code-2');
+
+  const showSignInButton = !isSignedIn && challengeIsCompleted;
 
   return (
     <div className='action-row-container'>
-      {renderButtons()}
+      {showSignInButton && (
+        <Button
+          data-cy='sign-in-button'
+          block={true}
+          href={`${apiLocation}/signin`}
+          className='btn-cta'
+        >
+          {t('learn.sign-in-save')}
+        </Button>
+      )}
+      <button
+        className={lowerJawButtonStyle}
+        data-cy='submit-lowerJaw-button'
+        onClick={tryToSubmitChallenge}
+        {...(!challengeIsCompleted && { 'aria-hidden': true })}
+        ref={submitButtonRef}
+      >
+        {t('buttons.submit-and-go')}
+      </button>
+      <button
+        className={lowerJawButtonStyle}
+        data-cy='check-lowerJaw-button'
+        onClick={tryToExecuteChallenge}
+        {...(challengeIsCompleted &&
+          !focusManagementCompleted && { tabIndex: -1, className: 'sr-only' })}
+        {...(focusManagementCompleted && { 'aria-hidden': true })}
+        ref={checkYourCodeButtonRef}
+      >
+        {checkButtonText}
+      </button>
```

Comment:
Could we use `aria-relevant` in submit button 🤔, to announce the change in the submit button component, instead of relaying on ref?

Reply:
`aria-relevant` is only used for live regions so it won't have any affect on the buttons. The refs on the Check Your Code and Submit buttons are being used for two things:
- To set focus on the Submit button
- To determine whether the Check Your Code button was clicked
I suppose we could give each of these buttons an `id` instead and use `getElementById` to do these two things, but I was under the impression that using `ref` to access these buttons is the proper way in react.

Reply:
Yeab it's the proper way to do it in react 👍, I was greedy to know how it would function with aria-relevant.